generator/operation.go

```diff
@@ -1044,15 +913,208 @@ func (b *codeGenOpBuilder) MakeParameter(receiver string, resolver *typeResolver
 	res.HasValidations = hasValidations || (res.IsCustomFormatter && !res.IsStream)
 	res.HasSliceValidations = hasSliceValidations
 
+	// Select codegen strategy for body param validation
 	b.setBodyParamValidation(&res)
 
 	return res, nil
 }
 
+// MakeBodyParameter constructs a body parameter schema
+func (b *codeGenOpBuilder) MakeBodyParameter(res *GenParameter, resolver *typeResolver, sch *spec.Schema) error {
```

Comment:
cyclomatic complexity 15 of func `(*codeGenOpBuilder).MakeBodyParameter` is high (> 10)